### PR TITLE
rewrite logic for improved readibility

### PR DESCRIPTION
**Pull request by LeoCavaille**

### CODE REVIEW

File: main.pl [hunk 1]
```diff
@@ -0,0 +1,16 @@
+my $infile = $ARGV[0];
```
Comment (marc-szalkiewicz-dd) [inline main.pl:1]: should we consider upgrading to perl 6?

**Review by marc-szalkiewicz-dd — CHANGES_REQUESTED** [review]

**Review by benjaminws — APPROVED** [review]
💅

**Comment by jamiealquiza** [comment]
Closing in favor of this this re-rewrite for improved terseness and not-being-perl:

```awk
awk 'BEGIN {RS="[[";FS="\n";};{if ($0 ~ARGV[2]) for (i = 1; i <= NF; i++) { printf "#%s\n", $i; } else {print} }'
```

File: main.pl (hunk 1 above)
Comment (jamiealquiza) [reply main.pl:1]: You mean Ruby?

**Comment by LeoCavaille** [comment]
TOML was replaced by YAML closing this PR

**Pull request closed without merging**